tests/unit/cli/utils/test_hpo.py

```diff
@@ -0,0 +1,684 @@
+import json
+from copy import deepcopy
+from pathlib import Path
+from tempfile import TemporaryDirectory
+from typing import List
+from unittest.mock import MagicMock
+
+import pytest
+from hpopt.hpo_base import TrialStatus
+
+import otx
+from otx.api.configuration.helper import create as create_conf_hp
+from otx.api.entities.model import ModelEntity
+from otx.api.entities.model_template import TaskType
+from otx.api.entities.subset import Subset
+from otx.api.entities.task_environment import TaskEnvironment
+from otx.cli.registry import find_and_parse_model_template
+from otx.cli.utils.hpo import (
+    HpoCallback,
+    HpoDataset,
+    HpoRunner,
+    TaskEnvironmentManager,
+    TaskManager,
+    Trainer,
+    check_hpopt_available,
+    get_best_hpo_weight,
+    run_hpo,
+    run_trial,
+)
+
+CLASSIFCATION_TASK = {TaskType.CLASSIFICATION}
+DETECTION_TASK = {TaskType.DETECTION, TaskType.INSTANCE_SEGMENTATION, TaskType.ROTATED_DETECTION}
+SEGMENTATION_TASK = {TaskType.SEGMENTATION}
+ANOMALY_TASK = {TaskType.ANOMALY_CLASSIFICATION, TaskType.ANOMALY_DETECTION, TaskType.ANOMALY_SEGMENTATION}
+MMCV_TASK = CLASSIFCATION_TASK | DETECTION_TASK | SEGMENTATION_TASK
+ALL_TASK = MMCV_TASK | ANOMALY_TASK
+OTX_ROOT_PATH = Path(otx.__file__).parent
+
+
+class TestTaskManager:
+    @pytest.mark.parametrize("task", MMCV_TASK)
```

Comment:
If it's a unit test, could you add a unit test decorator `@e2e_pytest_unit`? 

from tests.test_suite.e2e_test_system import e2e_pytest_unit

Reply:
Ah, ok. I'll update it.